.github/workflows/tests.yml

```
@@ -0,0 +1,201 @@
+  test-mcp-server:
+    runs-on: ubuntu-latest
+    
+    env:
+      MCP_API_KEY: test-api-key
+      JWT_SECRET_KEY: test-secret-key
+      ELASTICSEARCH_URL: http://localhost:9200
+      REDIS_URL: redis://localhost:6379
+      DATABASE_URL: postgresql://postgres:postgres@localhost:5432/mcp_test
+    
+    steps:
+    - uses: actions/checkout@v3
```

actionlint (1.7.7)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)